File: ruby/letter_inspection/src/application.rb
# rbs_inline: enabled

module LetterInspection
  class Application
    # @rbs source: String
    # @rbs target: String
    # @rbs return: bool
    def self.exactly_equal_size_and_included?(source:, target:)
      new(source, target).exactly_equal_size_and_included?
    end

    # @rbs source: String
    # @rbs target: String
    # @rbs return: void
    def initialize(source, target)
      @source = source
      @target = target
    end

    # @rbs return: bool
    def exactly_equal_size_and_included?
      sort_string(source) == sort_string(target)
    end

    private

    attr_reader :source, :target

    # @rbs str: String
    # @rbs return: String
    def sort_string(str)
      str.split('').sort.join('')
    end
  end
end

File: python/letter_inspection/src/application.py
class Application:
    def __init__(self, str_1: str, str_2: str) -> None:
        self.str_1 = str_1
        self.str_2 = str_2

    def exactly_equal_size_and_included(self) -> bool:
        return self.__sort_string__(self.str_1) == self.__sort_string__(self.str_2)

    # private

    def __sort_string__(self, str: str) -> str:
        return ''.join(sorted(str))
